packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt

```diff
@@ -0,0 +1,102 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+package com.facebook.react.uimanager
+
+import android.view.View
+import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.uimanager.annotations.ReactPropGroup
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+/**
+ * Test that verifies that spec of methods annotated with @ReactProp in {@link ReactShadowNode} is
+ * correct
+ */
+@RunWith(RobolectricTestRunner::class)
+@Ignore
+class ReactPropForShadowNodeSpecTest {
+
+  companion object {
+    private class BaseViewManager(
+      private val mShadowNodeClass: Class<out ReactShadowNode<ReactShadowNodeImpl>>
+    ): ViewManager<View?,ReactShadowNode<*>>() {
+      override fun getName(): String = "IgnoredName"
+
+      override fun createShadowNodeInstance(): ReactShadowNode<*>? = null
+
+      override fun getShadowNodeClass(): Class<out ReactShadowNode<ReactShadowNodeImpl>> = mShadowNodeClass
+
+      override fun createViewInstance(reactContext: ThemedReactContext): View = View(null)
+
+      override fun updateExtraData(root: View, extraData: Any?) {}
+
+    }
+
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testMethodWithWrongNumberOfParams() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactProp(name = "prop")
+        fun setterWithIncorrectNumberOfArgs(value: Boolean, anotherValue: Int) {}
+      }.javaClass
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testMethodWithTooFewParams() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl()  {
+        @ReactProp(name = "prop")
+        fun setterWithNoArgs() {}
+      }.javaClass
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testUnsupportedValueType() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactProp(name = "prop")
+        fun setterWithMap(value: Map<*, *>) {}
+      }.javaClass
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testGroupInvalidNumberOfParams() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactPropGroup(names = ["prop1", "prop2"])
+        fun setterWithTooManyParams(index: Int, value: Float, boolean: Boolean) {}
+      }.javaClass
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testGroupTooFewParams() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactPropGroup(names = ["props1", "prop2"])
+        fun setterWithTooFewParams(index: Int) {}
+      }.javaClass
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testGroupNoIndexParam() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactPropGroup(names = ["prop1", "prop2"])
+        fun setterWithNoIndexParam(value: Float, boolean: Boolean) {}
+      }::class.java
```

Comment:
Why `::class.java` here vs `.javaClass` in prior clauses?

Reply:
Silly mistake on my end. This has been updated